java/server/src/org/openqa/grid/web/Hub.java

```diff
@@ -77,27 +75,30 @@ public Registry getRegistry() {
     return registry;
   }
 
-  public Hub(GridHubConfiguration config) {
-    registry = Registry.newInstance(this, config);
+  public Hub(GridHubConfiguration gridHubConfiguration) {
+    registry = Registry.newInstance(this, gridHubConfiguration);
 
-    maxThread = config.getJettyMaxThreads();
-
-    if (config.getHost() != null) {
-      host = config.getHost();
+    config = gridHubConfiguration;
+    if (config.host != null) {
       isHostRestricted = true;
     } else {
       NetworkUtils utils = new NetworkUtils();
-      host = utils.getIp4NonLoopbackAddressOfThisMachine().getHostAddress();
+      config.host = utils.getIp4NonLoopbackAddressOfThisMachine().getHostAddress();
       isHostRestricted = false;
     }
-    this.port = config.getPort();
-
-    for (String s : config.getServlets()) {
-      Class<? extends Servlet> servletClass = ExtraServletUtil.createServlet(s);
-      if (servletClass != null) {
-        String path = "/grid/admin/" + servletClass.getSimpleName() + "/*";
-        log.info("binding " + servletClass.getCanonicalName() + " to " + path);
-        addServlet(path, servletClass);
+
+    if (config.port == null) {
+      config.port = 4444;
```

Comment:
was afraid someone might think so. I posit that the fields on these POJOs should be directly accessed, a getter /setter imply the variable could be manipulated in some way before returned. I want to discourage anyone potentially subclassing these Configuration classes and rather subclass the other more appropriate places in grid to change whatever behavior they intend on changing.